app/jobs/scheduled/redeliver_web_hook_events.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require "excon"
+
+module Jobs
+  class RedeliverWebHookEvents < ::Jobs::Scheduled
+    every 1.minute
+
+    sidekiq_options queue: "low"
+    sidekiq_options retry: false
+
+    REDELIVERED = "redelivered"
+    LIMIT = 20
+
+    def execute(args)
+      redelivery_events = RedeliveringWebhookEvent.includes(web_hook_event: :web_hook).limit(LIMIT)
+      redelivery_events.each do |redelivery_event|
+        begin
+          web_hook_event = redelivery_event.web_hook_event
+          web_hook = web_hook_event.web_hook
+
+          emitter = WebHookEmitter.new(web_hook, web_hook_event)
+          emitter.emit!(
+            headers: MultiJson.load(web_hook_event.headers),
+            body: web_hook_event.payload,
+          )
+
+          publish_webhook_event(web_hook_event, web_hook)
+          RedeliveringWebhookEvent.delete(redelivery_event)
+        rescue => e
+          Rails.logger.warn("Error redelivering web_hook_event #{web_hook_event.id}", e.inspect)
```

Comment:
Should use `Discourse.warn_exception` here, you can even pass in an "env" variable which could be the webhook event ID

Reply:
Also I do wonder if we also delete the `RedeliveringWebhookEvent` in this case? Instead of re-enqueuing it for up to 8 hours we could add a `REDELIVERY_FAILED` type and send it to the client like `publish_webhook_event`, then it is up to the admin to re-enqueue. Up to you, but this could be nicer.